src/StateSpaces.cpp

```diff
@@ -0,0 +1,194 @@
+        RAVELOG_ERROR("Can only use ContinuousJointsStateSpace with RobotProjectionEvaluator!");
+        return;
+    }
+
+    _robotStateSpace = robotStateSpace;
+}
+
+ContinuousJointsProjectionEvaluator::ContinuousJointsProjectionEvaluator(ompl::base::StateSpacePtr stateSpace) :
+    ProjectionEvaluator(stateSpace) {
+    ContinuousJointsStateSpace* robotStateSpace = dynamic_cast<ContinuousJointsStateSpace*>(stateSpace.get());
+
+    if (!robotStateSpace) {
+        RAVELOG_ERROR("Can only use ContinuousJointsStateSpace with RobotProjectionEvaluator!");
+        return;
+    }
+
+    _robotStateSpace = robotStateSpace;
+}
+
+ContinuousJointsProjectionEvaluator::~ContinuousJointsProjectionEvaluator() {
+
+}
+
+void ContinuousJointsProjectionEvaluator::setup() {
+    _projectionMatrix.mat = _projectionMatrix.ComputeRandom(_robotStateSpace->getDimension(), getDimension());
+    defaultCellSizes();
+    ProjectionEvaluator::setup();
+}
+
+void ContinuousJointsProjectionEvaluator::defaultCellSizes() {
+    cellSizes_.resize(getDimension());
+
+    for (size_t i = 0; i < getDimension(); i++) {
+        cellSizes_[i] = 0.5f;
+    }
+}
+
+/** \brief Return the dimension of the projection defined by this evaluator */
+unsigned int ContinuousJointsProjectionEvaluator::getDimension() const {
+    int dim = _robotStateSpace->getDimension();
+    if (dim <= 2) {
+        return dim;
+    }
+    else {
+        return (int)(log(_robotStateSpace->getDimension())) + 1;
+    }
+}
+
+/** \brief Compute the projection as an array of double values */
+void ContinuousJointsProjectionEvaluator::project(const ompl::base::State *state, ompl::base::EuclideanProjection &projection) const {
+    std::vector<double> values;
+    _robotStateSpace->copyToReals(values, state);
+    projection.resize(getDimension());
+    _projectionMatrix.project(values.data(), projection);
+}
+
+
+or_ompl::OrStateValidityChecker::OrStateValidityChecker(
+        const ompl::base::SpaceInformationPtr &si,
+        OpenRAVE::RobotBasePtr robot, std::vector<int> const &indices):
+    ompl::base::StateValidityChecker(si),
+    m_stateSpace(si->getStateSpace().get()),
+    m_env(robot->GetEnv()), m_robot(robot), m_indices(indices)
+{
+    resetStatistics();
+}
+
+bool or_ompl::OrStateValidityChecker::computeFk(const ompl::base::State *state, uint32_t checklimits) const
+{
+    std::vector<double> values;
+    m_stateSpace->copyToReals(values, state);
+    
+    BOOST_FOREACH(double v, values) {
+        if(std::isnan(v)) {
+            RAVELOG_ERROR("Invalid value in state.\n");
+            return false;
+        }
+    }
+    
+    m_robot->SetDOFValues(values, checklimits, m_indices);
+    return true;
+}
+
+bool or_ompl::OrStateValidityChecker::isValid(const ompl::base::State *state) const
+{
+    boost::chrono::steady_clock::time_point const tic
+       = boost::chrono::steady_clock::now();
+    
+    bool const collided = !computeFk(state, OpenRAVE::KinBody::CLA_Nothing)
```

Comment:
See above ... this PR just reorganizes code, but keeps the same behavior as the old code.  I do agree, however, that treating an FK failure (in the derived classes we have implemented so far, this can only currently happen due to the `nan` check above) as a collision seems a bit dubious.